src/core/lib/security/transport/security_handshaker.c

```diff
@@ -185,6 +188,58 @@ static void on_peer_checked(grpc_exec_ctx *exec_ctx, void *arg,
   security_handshaker_unref(exec_ctx, h);
 }
 
+static grpc_security_connector *find_lb_connector(grpc_exec_ctx *exec_ctx,
+                                                  grpc_channel_args *args) {
+  grpc_channel_security_connector *lb_channel_connector = NULL;
+  char *target = NULL;
+
+  const grpc_arg *target_info_arg =
+      grpc_channel_args_find(args, GRPC_ARG_GRPCLB_BALANCER_NAMES);
+  if (target_info_arg == NULL) goto done;
+
+  // To which address are we connecting?
+  grpc_resolved_address addr;
+  grpc_get_subchannel_address_arg(args, &addr);
+  GPR_ASSERT(grpc_sockaddr_to_string(&target, &addr, true /* normalize */));
+
+  const grpc_channel_targets_info *targets_info =
+      target_info_arg->value.pointer.p;
+  GPR_ASSERT(targets_info != NULL);
+
+  // Find the balancer name for the target.
+  const char *balancer_name =
+      grpc_channel_targets_info_find(targets_info, target);
+  if (balancer_name == NULL) {
+    gpr_log(GPR_ERROR, "Missing secure name for target address '%s'", target);
+    goto done;
+  }
+
+  // Get the LB channel's credentials
+  grpc_channel_credentials *lb_creds =
+      grpc_find_channel_credentials_in_args(args);
+  if (lb_creds == NULL) goto done;
+
+  // For LB channels, regenerate the security connector. The credentials used
+  // for the new security connection are stripped of call credentials and has
+  // knowledge about the balancer names, which are to be used as the "target"
+  // for the purposes of secure naming.
+  grpc_channel_args *new_args_from_connector = NULL;
+  const grpc_security_status security_status =
+      grpc_channel_credentials_create_security_connector(
```

Comment:
check_peer() doesn't currently have access to channel args. Are you also suggesting it be added? Its current signature is 
```
void grpc_security_connector_check_peer(grpc_exec_ctx* exec_ctx, grpc_security_connector* sc, tsi_peer peer, grpc_auth_context** auth_context, grpc_closure* on_peer_checked)
```

security handshaker invokes it as (from [here](https://github.com/grpc/grpc/blob/master/src/core/lib/security/transport/security_handshaker.c#L196))
```
grpc_security_connector_check_peer(exec_ctx, h->connector, peer,
                                     &h->auth_context, &h->on_peer_checked);
```

Two options
1. Simply add a new arg, passing h->args->args (<- preferred)
1. Pass in `h`, a `security_handshaker` instance and let the check_peer function read off it what it needs.